### ISSUE TITLE
Keep track of files installed by packages

### ISSUE THREAD

**Issue by Boris-Dimov** [FRUE-Project/fpkg#3]
Bc packages almost always contain more than a single file, would be good to have a place where to keep track of all the files installed by various packages.

For example: we create a dir at /etc/fpkg/; there, we will have configs etc, but also a subdir 'packages'; in this subdir, there will be a directory with the package name, which will hold a file named `fileslist` or sth, where on each line there will be the full path to the installed files from this package. 

I think this will make removal cleaner. Also, the package dir can hold other things like metadata.

**Comment by Kaper365** [comment]
Suggested resolution: ship a list of files and dirs that are going to be created with every fpkg file and it would serve as install and delete list

Suggestion: split created files into categories, that way stuff like configs won't be deleted because of accidental removal

**Comment by Boris-Dimov** [comment]
So sth like (on a single line):

FILENAME/DIRNAME IN TAR:PATH ON FILESYSTEM:TYPE

Assuming ':' is the delimeter.

**Comment by Kaper365** [comment]
I meant rather this kind of file:

files: {
file 1 (/home/${usr}/file1;
file 2 (/usr/bin/file2);
}


And with this categories I meant:

necessary {
file1
}

config {
file2 
}

misc {
file3
}

**Comment by Boris-Dimov** [comment]
Shell scripts work much easier with data on a single line, rather than JSON-style data.

We should settle on a file list format before going any further imo.

I propose the following:

There will be 2 lists of files - one for config files (not gonna be deleted in uninstall unless forced to) and one for regular files
They will reside in the tar, and will be copied to the package's folder in '/etc/fpkg/packages'
I propose that in the .tar they be named however the package maintainer likes, and the filenames will be given in the metadata
But once we copy them over, we'll rename them to sth standard like "configs_list" and "files_list"

Each like of the file will be as follows:

PATH_IN_TARBALL:PATH_ON_FILESYSTEM

Ez pz

**Comment by Kaper365** [comment]
I'd propose this layout:
```
| tar
|-- auto_install_script     //if necessary for e.g. config
|-- individual_files
|---- //stuff that goes e.g into ~/.config
|-- program_files
|---- binaries
|---- misc               // e.g. license
```
We'd enforce strict packaging practices in terms of naming of dirs and essential files

**Comment by Boris-Dimov** [comment]
How do you imagine that working with real filenames? I just can't comprehend what's a filename and what's a category description

**Comment by Kaper365** [comment]
Just like C programs have stuff like MAKEFILE or LICENSE that's just always there, we'd also have some fixed names (My example was rather provisional)

**Comment by Boris-Dimov** [comment]
Aha so these are the fixed names you propose

**Comment by Kaper365** [comment]
Not necessarily, we'll discuss the names after we'll set the way of creating of fpkgs
